packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -57,9 +63,25 @@ class DaemonCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    if (argResults['listen-on-tcp-port'] != null) {
+      final String port = stringArg('listen-on-tcp-port');
+      await _DaemonServer(
+        port: int.parse(port),
```

Comment:
This should probably be checked that it's a valid int.

```suggestion
      int port;
      try {
        port = int.parse(stringArg('listen-on-tcp-port'));
      } on FormatException catch (error) {
        throwToolExit('Invalid port for `--listen-on-tcp-port`: $error');
      }
      await _DaemonServer(
        port: port,
```